code/harness/DiskContents.h

```diff
@@ -0,0 +1,63 @@
+#ifndef DISK_CONTENTS_H
+#define DISK_CONTENTS_H
+
+#include <dirent.h>     /* Defines DT_* constants */
+#include <fcntl.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/mount.h>
+#include <sys/stat.h>
+#include <sys/syscall.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#include <fstream>
+#include <iostream>
+#include <string>
+#include <vector>
+#include <map>
+
+namespace fs_testing {
+
+class fileAttributes {
+public:
+  struct dirent* dir_attr;
+  struct stat* stat_attr;
+  std::string md5sum;
+
+  fileAttributes();
+  ~fileAttributes();
+
+  void set_dir_attr(struct dirent* a);
```

Comment:
generally I think C++ people tend to pass `struct dirent &a` instead of using pointers. If the compiler is halfway decent it should do copy elision on the parameter and you'll just magically have it

Reply:
ah, I see why it is `dirent *a`, though you could probably still do `struct stat &a` if you wanted.